src/main/java/org/folio/services/invoice/InvoiceFiscalYearsService.java

```diff
@@ -86,7 +86,7 @@ private Future<FiscalYearCollection> getFiscalYearsByFundIdsAndBudgets(List<Stri
     return fiscalYearService.getFiscalYearCollectionByQuery(query, requestContext)
       .map(fyCollection -> {
         if (fyCollection.getTotalRecords() == 0) {
-          log.error("getFiscalYearsByFundIdsAndBudgets(): no fiscal year left after filtering by date in invoice {}",
+          logger.error("getFiscalYearsByFundIdsAndBudgets(): no fiscal year left after filtering by date in invoice {}",
```

Comment:
logger.error("getFiscalYearsByInvoiceAndLines:: no fund related to the invoice with id {}", invoice.getId());
the common patter I have seem many places.

Reply:
I think these logs should be removed. They are followd by `throwCouldNotFindValidFiscalYearError` call which logs error and throws an exception which is logging 2 times by itself. Mostly we are logging same thing many times. Maybe we should try to find places where we do not log at all instead? I'm not sure...

Reply:
refactored so that `throwCouldNotFindValidFiscalYearError` logs more meaningful message and duplicates removed